scripts/safe_template_linter.py

```diff
@@ -1464,244 +1852,137 @@ def check_python_file_is_safe(self, file_contents, results):
-            'interpolate_end_index': interpolate_end_index,
-        }
+            The Python file contents with the encoding stripped.
+        """
+        # PEP-263 Provides Regex for Declaring Encoding
+        # Example: -*- coding: <encoding name> -*-
+        # This is only allowed on the first two lines, and it must be stripped
+        # before parsing, because we have already read into Unicode and the
+        # AST parser complains.
+        encoding_regex = re.compile(r"^[ \t\v]*#.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)")
+        encoding_match = encoding_regex.search(file_contents)
+        # If encoding comment not found on first line, search second line.
+        if encoding_match is None:
+            lines = StringLines(file_contents)
+            if lines.line_count() >= 2:
+                encoding_match = encoding_regex.search(lines.line_number_to_line(2))
+        # If encoding was found, strip it
+        if encoding_match is not None:
+            file_contents = file_contents.replace(encoding_match.group(), '#', 1)
+        return file_contents
 
-    def _check_format_html_strings_wrapped(self, strings, html_calls, results):
+    def _check_deprecated_display_name(self, file_contents, results):
         """
-        Checks that any string inside a format call that seems to contain HTML
-        is wrapped with a call to HTML().
+        Checks that the deprecated display_name_with_default_escaped is not
+        used. Adds violation to results if there is a problem.
 
         Arguments:
-            strings: A list of ParseStrings for each string inside the format()
-                call.
-            html_calls: A list of Expressions representing all of the HTML()
-                calls inside the format() call.
+            file_contents: The contents of the Python file
             results: A list of results into which violations will be added.
 
         """
-        html_strings = []
-        html_wrapped_strings = []
-        if strings is not None:
-            # find all strings that contain HTML
-            for string in strings:
-                if '<' in string.string:
-                    html_strings.append(string)
-                    # check if HTML string is appropriately wrapped
-                    for html_call in html_calls:
-                        if html_call.start_index < string.start_index < string.end_index < html_call.end_index:
-                            html_wrapped_strings.append(string)
-                            break
-            # loop through all unwrapped strings
-            for unsafe_string in set(html_strings) - set(html_wrapped_strings):
-                unsafe_string_expression = Expression(unsafe_string.start_index)
-                results.violations.append(ExpressionRuleViolation(
-                    Rules.python_wrap_html, unsafe_string_expression
-                ))
+        for match in re.finditer(r'\.display_name_with_default_escaped', file_contents):
```

Comment:
@nedbat: It is odd.  This was pre-existing code, and I needed to draw the line somewhere on the refactor.  I could add `TODO` comments, but I was trying to draw the line somewhere.  The same thing could be said for `python_custom_escape` below, but that gets slightly more complicated.  I may see how quickly I could take care of 1 or both.  Ugh. ;)